sdk/core/System.ClientModel/src/Pipeline/PipelineTransport.cs

```diff
@@ -67,51 +67,13 @@ private async ValueTask ProcessSyncOrAsync(PipelineMessage message, bool async)
         message.Response.SetIsError(ClassifyResponse(message));
         message.Response!.NetworkTimeout = networkTimeout;
 
-        if (message.Response!.ContentStream is null)
+        if (async)
         {
-            // No need to buffer if there is no content stream.
-            return;
+            await message.Response!.UpdateContentAsync(message.BufferResponse, userToken, joinedTokenSource).ConfigureAwait(false);
         }
-
-        if (!message.BufferResponse)
```

Comment:
Move this logic into Response so transport just makes a simple call to response.TryBufferContent(message.BufferContent)